src/main/java/com/google/sps/api/Utils.java

```diff
@@ -0,0 +1,82 @@
+// limitations under the License.
+
+
+package com.google.sps.api;
+
+import com.google.api.client.extensions.appengine.datastore.AppEngineDataStoreFactory;
+import com.google.api.client.extensions.appengine.http.UrlFetchTransport;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.Preconditions;
+import com.google.api.client.util.store.DataStoreFactory;
+import com.google.api.services.calendar.CalendarScopes;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Collections;
+
+import javax.servlet.http.HttpServletRequest;
+
+public class Utils {
+
+  /**
+   * Global instance of the {@link DataStoreFactory}. The best practice is to make it a single
+   * globally shared instance across your application.
+   */
+  private static final AppEngineDataStoreFactory DATA_STORE_FACTORY =
+      AppEngineDataStoreFactory.getDefaultInstance();
+  
+  /** Global instance of the HTTP transport. */
+  public static final HttpTransport HTTP_TRANSPORT = new UrlFetchTransport();
+
+  /** Global instance of the JSON factory. */
+  public static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+  private static GoogleClientSecrets clientSecrets = null;
+
+  /**
+   * Load the application's client secrets from the resources/client_secrets.json file
+   */
+  public static GoogleClientSecrets getClientCredential() throws IOException {
+    if (clientSecrets == null) {
+      clientSecrets = GoogleClientSecrets.load(JSON_FACTORY,
+          new InputStreamReader(Utils.class.getResourceAsStream("/client_secrets.json")));
+      Preconditions.checkArgument(!clientSecrets.getDetails().getClientId().startsWith("Enter ")
+          && !clientSecrets.getDetails().getClientSecret().startsWith("Enter "),
+          "Download client_secrets.json file from https://code.google.com/apis/console/"
+          + "?api=calendar into calendar-appengine-sample/src/main/resources/client_secrets.json");
+    }
+    return clientSecrets;
+  }
+
+  public static String getRedirectUri(HttpServletRequest req) {
+    GenericUrl url = new GenericUrl(req.getRequestURL().toString());
+    url.setRawPath("/oauth2callback");
+    return url.build();
+  }
+
+  /**
+   * Get a new OAuth2 authorization code flow for a request.
+   */
+  public static GoogleAuthorizationCodeFlow newFlow() throws IOException {
+    return new GoogleAuthorizationCodeFlow.Builder(HTTP_TRANSPORT, JSON_FACTORY,
+            getClientCredential(), Collections.singleton(CalendarScopes.CALENDAR_EVENTS)).setDataStoreFactory(
+            DATA_STORE_FACTORY).setAccessType("offline").build();
```

Comment:
offline should also be a constant. Maybe it's even exposed somewhere from the API code.